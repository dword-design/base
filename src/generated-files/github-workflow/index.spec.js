import proxyquire from '@dword-design/proxyquire'
import { outputFile } from 'fs-extra'
import withLocalTmpDir from 'with-local-tmp-dir'

export default {
  'job matrix': function () {
    expect(
      proxyquire('.', {
        '../../config': {
          nodeVersion: 14,
          useJobMatrix: true,
        },
      })
    ).toMatchSnapshot(this)
  },
  'no job matrix': function () {
    expect(
      proxyquire('.', {
        '../../config': {
          nodeVersion: 14,
        },
      })
    ).toMatchSnapshot(this)
  },
  'test environment variables': function () {
    return withLocalTmpDir(async () => {
      await outputFile(
        '.env.schema.json',
        { bar: {}, foo: {} } |> JSON.stringify
      )
      expect(
        proxyquire('.', {
          '../../config': {
            nodeVersion: 14,
          },
          './strategies/simple': proxyquire('./strategies/simple', {
            '../steps/test': proxyquire('./steps/test', {}),
          }),
        })
      ).toMatchSnapshot(this)
    })
  },
  testInContainer() {
    expect(
      proxyquire('.', {
        '../../config': {
          nodeVersion: 14,
          testInContainer: true,
          useJobMatrix: true,
        },
      })
    ).toMatchSnapshot(this)
  },
}
