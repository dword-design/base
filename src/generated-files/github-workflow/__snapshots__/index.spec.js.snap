// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index job matrix 1`] = `
Object {
  "jobs": Object {
    "cancel-existing": Object {
      "if": "!contains(github.event.head_commit.message, '[skip ci]')",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "uses": "rokroskar/workflow-run-cleanup-action@v0.3.2",
        },
      ],
    },
    "release": Object {
      "needs": "test",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v2",
          "with": Object {
            "lfs": true,
          },
        },
        Object {
          "uses": "actions/setup-node@v2",
          "with": Object {
            "node-version": 12,
          },
        },
        Object {
          "run": "git config --global user.email \\"actions@github.com\\"",
        },
        Object {
          "run": "git config --global user.name \\"GitHub Actions\\"",
        },
        Object {
          "run": "yarn --frozen-lockfile",
        },
        Object {
          "run": "yarn lint",
        },
        Object {
          "env": Object {
            "GITHUB_REPOSITORY": "\${{ secrets.GITHUB_REPOSITORY }}",
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "name": "Push changed files",
          "run": "yarn dw-ci push-changed-files",
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
            "NPM_TOKEN": "\${{ secrets.NPM_TOKEN }}",
          },
          "if": "github.ref == 'refs/heads/master'",
          "name": "Release",
          "run": "yarn semantic-release",
        },
      ],
    },
    "test": Object {
      "needs": "cancel-existing",
      "runs-on": "\${{ matrix.os }}",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v2",
          "with": Object {
            "fetch-depth": 0,
            "lfs": true,
          },
        },
        Object {
          "uses": "actions/setup-node@v2",
          "with": Object {
            "node-version": "\${{ matrix.node }}",
          },
        },
        Object {
          "run": "yarn --frozen-lockfile",
        },
        Object {
          "run": "yarn test",
        },
        Object {
          "if": "failure()",
          "uses": "actions/upload-artifact@v2",
          "with": Object {
            "path": "**/__image_snapshots__/__diff_output__",
          },
        },
        Object {
          "run": "yarn checkUnknownFiles",
        },
        Object {
          "if": "matrix.os == 'ubuntu-latest' && matrix.node == 12",
          "uses": "codecov/codecov-action@v1",
          "with": Object {
            "fail_ci_if_error": true,
            "token": "\${{ secrets.CODECOV_TOKEN }}",
          },
        },
      ],
      "strategy": Object {
        "matrix": Object {
          "exclude": Array [
            Object {
              "node": 10,
              "os": "macos-latest",
            },
            Object {
              "node": 10,
              "os": "windows-latest",
            },
          ],
          "node": Array [
            10,
            12,
          ],
          "os": Array [
            "macos-latest",
            "windows-latest",
            "ubuntu-latest",
          ],
        },
      },
    },
  },
  "name": "build",
  "on": Object {
    "push": Object {
      "branches": Array [
        "**",
      ],
    },
  },
}
`;

exports[`index no job matrix 1`] = `
Object {
  "jobs": Object {
    "build": Object {
      "if": "!contains(github.event.head_commit.message, '[skip ci]')",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "uses": "rokroskar/workflow-run-cleanup-action@v0.3.2",
        },
        Object {
          "uses": "actions/checkout@v2",
          "with": Object {
            "fetch-depth": 0,
            "lfs": true,
          },
        },
        Object {
          "uses": "actions/setup-node@v2",
          "with": Object {
            "node-version": 12,
          },
        },
        Object {
          "run": "git config --global user.email \\"actions@github.com\\"",
        },
        Object {
          "run": "git config --global user.name \\"GitHub Actions\\"",
        },
        Object {
          "run": "yarn --frozen-lockfile",
        },
        Object {
          "run": "yarn test",
        },
        Object {
          "if": "failure()",
          "uses": "actions/upload-artifact@v2",
          "with": Object {
            "path": "**/__image_snapshots__/__diff_output__",
          },
        },
        Object {
          "run": "yarn checkUnknownFiles",
        },
        Object {
          "uses": "codecov/codecov-action@v1",
          "with": Object {
            "fail_ci_if_error": true,
            "token": "\${{ secrets.CODECOV_TOKEN }}",
          },
        },
        Object {
          "env": Object {
            "GITHUB_REPOSITORY": "\${{ secrets.GITHUB_REPOSITORY }}",
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "name": "Push changed files",
          "run": "yarn dw-ci push-changed-files",
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
            "NPM_TOKEN": "\${{ secrets.NPM_TOKEN }}",
          },
          "if": "github.ref == 'refs/heads/master'",
          "name": "Release",
          "run": "yarn semantic-release",
        },
      ],
    },
  },
  "name": "build",
  "on": Object {
    "push": Object {
      "branches": Array [
        "**",
      ],
    },
  },
}
`;

exports[`index test environment variables 1`] = `
Object {
  "jobs": Object {
    "build": Object {
      "if": "!contains(github.event.head_commit.message, '[skip ci]')",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "uses": "rokroskar/workflow-run-cleanup-action@v0.3.2",
        },
        Object {
          "uses": "actions/checkout@v2",
          "with": Object {
            "fetch-depth": 0,
            "lfs": true,
          },
        },
        Object {
          "uses": "actions/setup-node@v2",
          "with": Object {
            "node-version": 12,
          },
        },
        Object {
          "run": "git config --global user.email \\"actions@github.com\\"",
        },
        Object {
          "run": "git config --global user.name \\"GitHub Actions\\"",
        },
        Object {
          "run": "yarn --frozen-lockfile",
        },
        Object {
          "env": Object {
            "TEST_BAR": "\${{ secrets.TEST_BAR }}",
            "TEST_FOO": "\${{ secrets.TEST_FOO }}",
          },
          "run": "yarn test",
        },
        Object {
          "if": "failure()",
          "uses": "actions/upload-artifact@v2",
          "with": Object {
            "path": "**/__image_snapshots__/__diff_output__",
          },
        },
        Object {
          "run": "yarn checkUnknownFiles",
        },
        Object {
          "uses": "codecov/codecov-action@v1",
          "with": Object {
            "fail_ci_if_error": true,
            "token": "\${{ secrets.CODECOV_TOKEN }}",
          },
        },
        Object {
          "env": Object {
            "GITHUB_REPOSITORY": "\${{ secrets.GITHUB_REPOSITORY }}",
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "name": "Push changed files",
          "run": "yarn dw-ci push-changed-files",
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
            "NPM_TOKEN": "\${{ secrets.NPM_TOKEN }}",
          },
          "if": "github.ref == 'refs/heads/master'",
          "name": "Release",
          "run": "yarn semantic-release",
        },
      ],
    },
  },
  "name": "build",
  "on": Object {
    "push": Object {
      "branches": Array [
        "**",
      ],
    },
  },
}
`;

exports[`index testInContainer 1`] = `
Object {
  "jobs": Object {
    "cancel-existing": Object {
      "if": "!contains(github.event.head_commit.message, '[skip ci]')",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "uses": "rokroskar/workflow-run-cleanup-action@v0.3.2",
        },
      ],
    },
    "release": Object {
      "needs": "test",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v2",
          "with": Object {
            "lfs": true,
          },
        },
        Object {
          "uses": "actions/setup-node@v2",
          "with": Object {
            "node-version": 12,
          },
        },
        Object {
          "run": "git config --global user.email \\"actions@github.com\\"",
        },
        Object {
          "run": "git config --global user.name \\"GitHub Actions\\"",
        },
        Object {
          "run": "yarn --frozen-lockfile",
        },
        Object {
          "run": "yarn lint",
        },
        Object {
          "env": Object {
            "GITHUB_REPOSITORY": "\${{ secrets.GITHUB_REPOSITORY }}",
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "name": "Push changed files",
          "run": "yarn dw-ci push-changed-files",
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
            "NPM_TOKEN": "\${{ secrets.NPM_TOKEN }}",
          },
          "if": "github.ref == 'refs/heads/master'",
          "name": "Release",
          "run": "yarn semantic-release",
        },
      ],
    },
    "test": Object {
      "needs": "cancel-existing",
      "runs-on": "\${{ matrix.os }}",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v2",
          "with": Object {
            "fetch-depth": 0,
            "lfs": true,
          },
        },
        Object {
          "uses": "actions/setup-node@v2",
          "with": Object {
            "node-version": "\${{ matrix.node }}",
          },
        },
        Object {
          "run": "yarn --frozen-lockfile",
        },
        Object {
          "run": "yarn test",
        },
        Object {
          "if": "failure()",
          "uses": "actions/upload-artifact@v2",
          "with": Object {
            "path": "**/__image_snapshots__/__diff_output__",
          },
        },
        Object {
          "run": "yarn checkUnknownFiles",
        },
        Object {
          "if": "matrix.os == 'ubuntu-latest' && matrix.node == 12",
          "uses": "codecov/codecov-action@v1",
          "with": Object {
            "fail_ci_if_error": true,
            "token": "\${{ secrets.CODECOV_TOKEN }}",
          },
        },
      ],
      "strategy": Object {
        "matrix": Object {
          "node": Array [
            10,
            12,
          ],
          "os": Array [
            "ubuntu-latest",
          ],
        },
      },
    },
  },
  "name": "build",
  "on": Object {
    "push": Object {
      "branches": Array [
        "**",
      ],
    },
  },
}
`;
